# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserBridgeHost.cpp
# Commit: e65cb2d4c5a5
# Full Hash: e65cb2d4c5a55e3049922df02af643337b7a58b2
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 04:41:59
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 16 - Cycle collect RemoteBrowser and support weak references in BrowserHost. r=nika
#   
#   BrowserParent is cycle collected and supported weak references, so this commit adds support
#   for these things to BrowserHost.
#   
# ==============================================================================

diff -r 99f971a02d87 -r e65cb2d4c5a5 dom/ipc/BrowserBridgeHost.cpp
--- a/dom/ipc/BrowserBridgeHost.cpp	Wed May 08 16:12:26 2019 -0500
+++ b/dom/ipc/BrowserBridgeHost.cpp	Tue May 14 16:28:10 2019 -0500
@@ -9,6 +9,15 @@
 namespace mozilla {
 namespace dom {
 
+NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(BrowserBridgeHost)
+  NS_INTERFACE_MAP_ENTRY(nsISupports)
+NS_INTERFACE_MAP_END
+
+NS_IMPL_CYCLE_COLLECTION(BrowserBridgeHost)
+
+NS_IMPL_CYCLE_COLLECTING_ADDREF(BrowserBridgeHost)
+NS_IMPL_CYCLE_COLLECTING_RELEASE(BrowserBridgeHost)
+
 BrowserBridgeHost::BrowserBridgeHost(BrowserBridgeChild* aChild)
     : mBridge(aChild) {}
 